Sources/Build/BuildPlan.swift

```diff
@@ -197,6 +192,11 @@ public final class ClangTargetDescription {
         args += additionalFlags
         args += moduleCacheArgs
         args += buildParameters.flags.cCompilerFlags
+
+        // Add extra C++ flags if this target contains C++ files.
+        if clangTarget.isCXX {
```

Comment:
Please document the ordering constraint too.

Reply:
I've added comments for the change in this PR, but also added comments for the previous PR (https://github.com/apple/swift-package-manager/pull/1365) which covered -Xswiftc, -Xlinker and -Xcc to highlight the fact that those options should go last to enable user overrides of the generated args.